plugins/outputs/azure_data_explorer/azure_data_explorer_test.go

```diff
@@ -18,217 +16,280 @@ import (
+			createTables:              false,
+			client:                    fakeClient,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap,
 		},
 		{
-			name:         "Error in Mgmt",
-			inputMetric:  testutil.MockMetrics(),
-			createTables: true,
-			client: &fakeClient{
-				queries: make([]string, 0),
-				internalMgmt: func(f *fakeClient, ctx context.Context, db string, query kusto.Stmt, options ...kusto.MgmtOption) (*kusto.RowIterator, error) {
-					return nil, errors.New("Something went wrong")
-				},
-			},
-			createIngestor:  createFakeIngestor,
-			metricsGrouping: tablePerMetric,
-			expected: map[string]interface{}{
-				"metricName": "test1",
-				"fields": map[string]interface{}{
-					"value": 1.0,
-				},
-				"tags": map[string]interface{}{
-					"tag1": "value1",
-				},
-				"timestamp": float64(time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC).UnixNano() / int64(time.Second)),
-			},
-			expectedWriteError: "creating table for \"test1\" failed: Something went wrong",
+			name:                      "SingleTable metric grouping type",
+			inputMetric:               mockMetrics,
+			createTables:              true,
+			client:                    fakeClient,
+			metricsGrouping:           singleTable,
+			tableNameToExpectedResult: expectedResultMap,
 		},
 		{
-			name:         "SingleTable metric grouping type",
-			inputMetric:  testutil.MockMetrics(),
-			createTables: true,
-			client: &fakeClient{
-				queries: make([]string, 0),
-				internalMgmt: func(f *fakeClient, ctx context.Context, db string, query kusto.Stmt, options ...kusto.MgmtOption) (*kusto.RowIterator, error) {
-					f.queries = append(f.queries, query.String())
-					return &kusto.RowIterator{}, nil
-				},
-			},
-			createIngestor:  createFakeIngestor,
-			metricsGrouping: singleTable,
-			expected: map[string]interface{}{
-				"metricName": "test1",
-				"fields": map[string]interface{}{
-					"value": 1.0,
-				},
-				"tags": map[string]interface{}{
-					"tag1": "value1",
-				},
-				"timestamp": float64(time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC).UnixNano() / int64(time.Second)),
-			},
+			name:                      "Valid metric managed ingestion",
+			inputMetric:               mockMetrics,
+			createTables:              true,
+			client:                    fakeClient,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap,
+			ingestionType:             managedIngestion,
+		},
+		{
+			name:                      "Table per metric type",
+			inputMetric:               mockMetricsMulti,
+			createTables:              true,
+			client:                    fakeClient,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap2,
 		},
 	}
 
 	for _, tC := range testCases {
 		t.Run(tC.name, func(t *testing.T) {
 			serializer, err := telegrafJson.NewSerializer(time.Second, "", "")
```

Comment:
you might want to parallel tests here and elsewhere (t.Parallel at the start of the function, but don't forget to pin the test case - https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721)